AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/StorageEngineAdapter+SQLite.swift

```diff
@@ -223,6 +226,23 @@ final class SQLiteStorageEngineAdapter: StorageEngineAdapter {
             try transactionBlock()
         }
     }
+
+    func clear(completion: @escaping DataStoreCallback<Void>) {
+        guard let dbFilePath = dbFilePath else {
+            log.error("Attempt to clear DB, but file path was empty")
+            completion(.failure(causedBy: DataStoreError.invalidDatabase(path: "Database path not set", nil)))
+            return
+        }
+        connection = nil
```

Comment:
I don't remember the SQLite API, but is there a cancel or teardown that we should invoke before releasing our connection?

Reply:
I looked in the following link, but didn't see anything:
https://github.com/stephencelis/SQLite.swift/blob/master/Documentation/Index.md

Hmm but, the way I arrived at this approach (`connection = nil`) is that i noticed that we needed to a call to `sqlite3_close()` -- which seems to be in the `deinit` of the `Connection` class.  So I figured this was how to 'teardown' the connection.

I'm going to leave this as-is for now.